src/dreyfus_index_updater.erl

```diff
@@ -78,14 +78,27 @@ load_docs(FDI, {I, IndexPid, Db, Proc, Total, LastCommitTime}=Acc) ->
             Fields1 = [list_to_tuple(Field) || Field <- Fields],
             case Fields1 of
                 [] -> ok = clouseau_rpc:delete(IndexPid, Id);
-                _  -> ok = clouseau_rpc:update(IndexPid, Id, Fields1)
+                _  -> 
+                    Fields2 = maybe_add_partition_key(DbName, Id, Fields1),
+                    ok = clouseau_rpc:update(IndexPid, Id, Fields2)
             end
     end,
     %% Force a commit every minute
     case timer:now_diff(Now = now(), LastCommitTime) >= 60000000 of
         true ->
             ok = clouseau_rpc:commit(IndexPid, Seq),
-            {ok, {I+1, IndexPid, Db, Proc, Total, Now}};
+            {ok, {I+1, IndexPid, Db, Proc, Total, Now, DbName}};
         false ->
             {ok, setelement(1, Acc, I+1)}
     end.
+
+
+maybe_add_partition_key(Db, Id, Fields) ->
+    case mem3:is_partitioned(Db) of 
+        true ->
+            PK = hd(binary:split(Id, <<":">>)),
+            Fields1 = [{<<"_partition">>, PK, {[]}} | Fields],
```

Comment:
If I remember correctly, the tuples were just {Key, Value} pairs that were going to be indexed. So for partitioning work, you are adding a new field ```<<"_partition">>``` with value ```PK```. What is the ```{[]}``` for?